### ISSUE TITLE
The right way to get latest version of a package 

### ISSUE THREAD

**Issue by coocheenin** [FedericoCeratto/nim-package-directory#18]
Hi,

I have notice the "new" version badge. I have looked into the code. What's magic!
I mean [this request](https://github.com/FedericoCeratto/nim-package-directory/blob/master/github.nim#L48). According github APIv3 docs it's the only way [to get the latest release version](https://developer.github.com/v3/repos/releases/#get-the-latest-release), BUT...

**In real life the most packages don't have release version.** Each time a repo is tagged, a new source code bundle will automatically show up on the github "Release" page. However, these automatic "releases" will not appear via the API. It's because I wrote about magic...

**I think it's more correct from a religious point of view to get the package version from a `.nimble` file. What do you think about?**

For example, the following code can be placed in a `except` [section](https://github.com/FedericoCeratto/nim-package-directory/blob/master/github.nim#L52):

```nim
let dotnimble = "https://raw.githubusercontent.com/" & owner_repo_name & "/master/" & repo & ".nimble"
let r = newHttpClient().get(dotnimble)
if r.code == Http200 :
  for line in splitLines(r.body) :
    if line.startsWith("version"): pkg["github_latest_version"] = split(r.body, "\"")[1]
```

Maybe I'm wrong.

**Comment by euantorano** [comment]
Note that a repository could have a different name to the main nimble file.

For example, take my repository [`euantorano/sever.nim`](https://github.com/euantorano/semver.nim) where the `.nimble` file name is actually [`semver.nimble`](https://github.com/euantorano/semver.nim).

I believe that the eventual plan was for the nimble directory to actually be the package source, rather than a large JSON file stored on GitHub. In that cause, authors would be submitting packages and we could probably ask them to provide the path to the `.nimble` file on submission.

**Comment by GULPF** [comment]
Note that .nimble files are not declarative, so extracting the version from them is not that simple. See e.g nimbles [.nimble file](https://github.com/nim-lang/nimble/blob/master/nimble.nimble), where the version number is imported from another module.

**Comment by coocheenin** [comment]
@euantorano Ok, but which purpose for naming like so?
(repo and package with different names)

And by the way, your repo has **no releases** yet.
Attempt to request `curl -i https://api.github.com/repos/euantorano/semver.nim/releases/latest` will raise 404.

**Comment by coocheenin** [comment]
> Note that .nimble files are not declarative, so extracting the version from them is not that simple. See e.g nimbles [.nimble file](https://github.com/nim-lang/nimble/blob/master/nimble.nimble), where the version number is imported from another module.

Yes, right. But I suggest a solution (in example) for cases where there is **no real release** (only tagged automatic release). I don't see in sources extracting the tag version (only [release extracting request](https://github.com/FedericoCeratto/nim-package-directory/blob/master/github.nim#L48)). And sadly, we can't assume that the tags are in chronological order.

**Comment by euantorano** [comment]
@coocheenin Yes, because Nimble doesn't actuallly use GitHub releases when it does version reoslution itself - it always looks at the `.nimble` file for the package. It decides upon this file's name based upon the name of the project in the package repository JSON.

Regarding why I have a different name for the repository and the project, most of my Nim projects have a name ending in `.nim` to allow me to easily filter them.

**Comment by coocheenin** [comment]
@euantorano I've understand. The only open question: Does it make sense to add code, typed above, to `fetch_github_versions` as I've described in issue's [UPDATE] section.

**Comment by euantorano** [comment]
Yes, it definitely needs improving in some way. I’m not sure what the best option is personally.

**Comment by coocheenin** [comment]
> Yes, it definitely needs improving in some way. I’m not sure what the best option is personally.

Can I make a PR then? With additional check whether the target a number is (to avoid situation described by @GULPF).

**Comment by coocheenin** [comment]
I did it in async style, with additional `isDigit()` check: 368b89f3c84ce78fabcb1cebfbbab6f2d95d2e28
But i have no idea about:
> ci/circleci Expected — Waiting for status to be reported

**Comment by FedericoCeratto** [comment]
GitHub's API does not provide the latest version anymore and it was also providing it based on date rather than a semantic on version numbers.
Fixed in https://github.com/FedericoCeratto/nim-package-directory/commit/f2f23d2f7ec1caf89d131d1d7282c0b8c61b0305
